core/src/sample_performance_service.rs

```diff
@@ -51,18 +51,20 @@ impl SamplePerformanceService {
         blockstore: &Arc<Blockstore>,
         exit: Arc<AtomicBool>,
     ) {
-        let (bank, highest_slot) = {
-            let forks = bank_forks.read().unwrap();
-            (forks.root_bank(), forks.highest_slot())
-        };
-
-        // Store the absolute transaction counts to that we can compute the
-        // difference between these values at points in time to figure out
-        // how many transactions occurred in that timespan.
-        let mut snapshot = SamplePerformanceSnapshot {
-            num_transactions: bank.transaction_count(),
-            num_non_vote_transactions: bank.non_vote_transaction_count_since_restart(),
-            highest_slot,
+        let mut snapshot = {
+            let (bank, highest_slot) = {
+                let forks = bank_forks.read().unwrap();
+                (forks.root_bank(), forks.highest_slot())
+            };
```

Comment:
While this code is going away, nested scope does not make that much sense anymore.
I've introduced it to remove an explicit `drop(forks)`.
But now it will be dropped after the next statement, even if the nested scope is gone.
It would be more code churn, though.

Reply:
Yeah, good point. Plus I think it will actually be easier to apply to v1.14 if I remove the nest. Let me rework.